src/java/main/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -310,6 +313,9 @@ public void respondToServer(byte[] serverToken, ClientCnxn cnxn) {
             // TODO: introspect about runtime environment (such as jaas.conf)
             saslState = SaslState.FAILED;
             throw new SaslException("Error in authenticating with a Zookeeper Quorum member: the quorum member's saslToken is null.");
+        } else if (new String(saslToken).equals(AUTHENTICATION_FAILED)) {
```

Comment:
Not setting an explicit charset is a code smell, in this case we are using only chars so it won't be a gread deal, but it is better ti have clean code